pkgs/development/misc/qmk/firmware.nix

```diff
@@ -0,0 +1,68 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, avrgcc
+, avrbinutils
+, gcc-arm-embedded
+, gcc-armhf-embedded
+, teensy-loader-cli
+, dfu-programmer
+, dfu-util
+}:
+
+stdenv.mkDerivation rec {
+  pname = "qmk_firmware";
+  version = "0.11.49";
+
+  src = fetchFromGitHub {
+    owner = "qmk";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-E/oe5u6F9pxvavRLlrXmr9kCc6pSdSmQgtg4a3wbIuA=";
+    fetchSubmodules = true;
+  };
+
+  postPatch = ''
+    substituteInPlace tmk_core/arm_atsam.mk \
+      --replace arm-none-eabi arm-none-eabihf
+    rm keyboards/handwired/frenchdev/rules.mk 
+    rm keyboards/dk60/rules.mk
+  '';
+
+  nativeBuildInputs = [
+    avrgcc
+    avrbinutils
+    gcc-arm-embedded
+    gcc-armhf-embedded
+    teensy-loader-cli
+    dfu-programmer
+    dfu-util
+  ];
+
+  buildFlags = [ "all:default" ];
+
+  NIX_CFLAGS_COMPILE = "-Wno-error";
+
+  doCheck = true;
+
+  checkTarget = "test:all";
+
+  installPhase = ''
+    mkdir $out
```

Comment:
You are sure about this? This shouldn't be preInstall?

Reply:
Seriously I don't know. I just rewrote the expression, but when I tried to test it, the GCC cross compiler didn't compile.